File: JPA/shop/src/main/java/me/kzv/shop/domain/Member.java
package me.kzv.shop.domain;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="member_id")
    private Long id;

    private String name;

    @Embedded
    private Address address;

    @OneToMany(mappedBy = "member") // order 클래스의 member 가 주인이다.
    private List<Order> orders = new ArrayList<>();
}

File: toy/e-commerce/src/main/kotlin/me/kzv/shop/entity/Member.kt
package me.kzv.shop.entity

import me.kzv.shop.controller.dto.MemberDto
import me.kzv.shop.entity.constant.Role
import org.springframework.security.crypto.password.PasswordEncoder
import javax.persistence.*

@Entity
class Member(

    @Id
    @Column(name = "member_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var name: String,

    var email: String,

    var password: String,

    var address: String,

    @Enumerated(EnumType.STRING)
    var role: Role,

    ) : BaseEntity()

fun createMember(dto: MemberDto, passwordEncoder: PasswordEncoder) = Member(
    name = dto.name,
    email = dto.email,
    address = dto.address,
    password = passwordEncoder.encode(dto.password),
    role = Role.ADMIN,
)
